docs/index.asciidoc

```diff
@@ -88,6 +88,14 @@ will be similar to events directly indexed by {plugin-uc} into Elasticsearch.
 endif::[]
 
 
+[id="plugins-{type}s-{plugin}-memory"]
+===== Memory usage
+
+This plugin uses direct memory which is not included in the heap memory.
+By default, it is the same as the heap size (Xmx), for example, setting 10GB Xmx means setting 10GB of direct memory.
+Make sure heap and direct memory does not exceed the total size of the server. You can set the amount of direct memory with -XX:MaxDirectMemorySize.
```

Comment:
Maybe add a link to the [jvm settings](https://www.elastic.co/guide/en/logstash/current/jvm-settings.html) doc in logstash core?

We should also update that document too, along the lines of this [issue](https://github.com/elastic/logstash/issues/12622)

Reply:
Adding a link sounds good. Users mostly come across direct memory with beats-input. I will update the JVM settings page. Thanks for the pointer